src/Compilers/CSharp/Portable/Binder/Semantics/Conversions/ConversionsBase.cs

```diff
@@ -168,6 +168,33 @@ public Conversion ClassifyConversionFromExpressionType(TypeSymbol source, TypeSy
             return ClassifyConversionFromType(source, destination, ref useSiteDiagnostics);
         }
 
+        private bool TryGetVoidConversion(TypeSymbol source, TypeSymbol destination, out Conversion conversion)
+        {
+            var sourceIsVoid = source?.SpecialType == SpecialType.System_Void;
+            var destIsVoid = destination.SpecialType == SpecialType.System_Void;
+
+            // 'void' is not supposed to be able to convert to or from anything, but in practice,
+            // a lot of code depends on checking whether an expression of type 'void' is convertible to 'void'.
+            // (e.g. for an expression lambda which returns void).
+            // Therefore we allow an identity conversion between 'void' and 'void'.
+            if (sourceIsVoid && destIsVoid)
+            {
+                conversion = Conversion.Identity;
+                return true;
+            }
+
+            // If exactly one of source or destination is of type 'void' then no conversion may exist.
+            if (sourceIsVoid || destIsVoid)
```

Comment:
>sourceIsVoid || destIsVoid [](start = 16, length = 26)

It looks like this method returns true for `source == null && destIsVoid`. Is there a test for that case?

Reply:
I've played with a few scenarios such as the following:

```cs
class C
{
    void M(System.Action a) {}
    void M2()
    {
        M(() => null);
    }
}
```

where I suspect the compiler might check convertibility of an expression with no type to an expression with a void type. But I wasn't able to get the compiler to emit any diagnostics related to the failed conversion, or enter the `TryGetVoidConversion` method, or do anything else that felt clearly "testable".